src/backend/sql/gnc-backend-sql.h

```diff
@@ -372,23 +372,23 @@ struct GncSqlColumnInfo
     GncSqlColumnInfo (std::string&& name, GncSqlBasicColumnType type,
                       unsigned int size = 0, bool unicode = false,
                       bool autoinc = false, bool primary = false,
-                      bool null_allowed = false) :
+                      bool not_null = false) :
```

Comment:
Yes. Making not_null the default blew up because there are far more columns that need to accept null values than there are ones that shouldn't. Note as well that this code is gone; the  GncSqlColumnInfo ctor now takes an or'd set of flags that correspond directly to the flags one passes to CREATE TABLE.